ChiModules/LinearBoltzmannSolver/Acceleration/diffusion_mip_02a_assembleAand_b.cc

```diff
@@ -0,0 +1,377 @@
+            }//for fj
+          }//for fi
+
+          //========================= Assemble gradient terms
+          // For the following comments we use the notation:
+          // Dk = 0.5* n dot nabla bk
+
+          // 0.5*D* n dot (b_j^+ - b_j^-)*nabla b_i^-
+          for (int i=0; i<num_nodes; i++)
+          {
+            const int64_t imap = m_sdm.MapDOF(cell, i, m_uk_man, 0, g);
+
+            for (int fj=0; fj<num_face_nodes; fj++)
+            {
+              const int jm = cell_mapping.MapFaceNode(f,fj);      //j-minus
+              const int jp = MapFaceNodeDisc(cell, adj_cell, cc_nodes, ac_nodes,
+                                             f, acf, fj);         //j-plus
+              const int64_t jmmap = m_sdm.MapDOF(cell,jm,m_uk_man,0,g);
+              const int64_t jpmap = m_sdm.MapDOF(adj_cell,jp,m_uk_man,0,g);
+
+              chi_mesh::Vector3 vec_aij;
+              for (size_t qp : fqp_data.QuadraturePointIndices())
+                vec_aij +=
+                  fqp_data.ShapeValue(jm, qp) * fqp_data.ShapeGrad(i, qp) *
+                  fqp_data.JxW(qp);
+              const double aij = -0.5*Dg*n_f.Dot(vec_aij);
+
+              MatSetValue(m_A, imap, jmmap, aij, ADD_VALUES);
+              MatSetValue(m_A, imap, jpmap,-aij, ADD_VALUES);
+            }//for fj
+          }//for i
+
+          // 0.5*D* n dot (b_i^+ - b_i^-)*nabla b_j^-
+          for (int fi=0; fi<num_face_nodes; fi++)
+          {
+            const int im = cell_mapping.MapFaceNode(f,fi);       //i-minus
+            const int ip = MapFaceNodeDisc(cell,adj_cell,cc_nodes,ac_nodes,
+                                           f,acf,fi);            //i-plus
+            const int64_t immap = m_sdm.MapDOF(cell,im,m_uk_man,0,g);
+            const int64_t ipmap = m_sdm.MapDOF(adj_cell,ip,m_uk_man,0,g);
+
+            for (int j=0; j<num_nodes; j++)
+            {
+              const int64_t jmap = m_sdm.MapDOF(cell, j, m_uk_man, 0, g);
+
+              chi_mesh::Vector3 vec_aij;
+              for (size_t qp : fqp_data.QuadraturePointIndices())
+                vec_aij +=
+                  fqp_data.ShapeValue(im, qp) * fqp_data.ShapeGrad(j, qp) *
+                  fqp_data.JxW(qp);
+              const double aij = -0.5*Dg*n_f.Dot(vec_aij);
+
+              MatSetValue(m_A,immap,jmap, aij,ADD_VALUES);
+              MatSetValue(m_A,ipmap,jmap,-aij,ADD_VALUES);
+            }//for j
+          }//for fi
+
+        }//internal face
+        else
+        {
+          auto bc = DefaultBCDirichlet;
+          try {bc = m_bcs.at(face.neighbor_id);}
+          catch (const std::out_of_range& oor)
+          {throw std::logic_error(fname + ": unmapped boundary id.");}
+
+          if (bc.type == BCType::DIRICHLET)
+          {
+            const double bc_value = bc.values[0];
+
+            //========================= Compute kappa
+            double kappa = 1.0;
+            if (cell.Type() == chi_mesh::CellType::SLAB)
```

Comment:
on the BC, the constant is simply double what it is inside, so that could still work as something supplied by the user